piv/piv.go

```diff
@@ -846,3 +846,121 @@ func ykSetProtectedMetadata(tx *scTx, key [24]byte, m *Metadata) error {
 	}
 	return nil
 }
+
+// CardID is the Card Holder Unique Identifier with settable GUID
+// Raw contains the whole object
+// GUID contains the Card Universally Unique Identifier.
+type CardID struct {
+	Raw  []byte
+	GUID [16]byte
+}
+
+// CardID returns the card CHUID with the GUID extracted
+func (yk *YubiKey) CardID() (CardID, error) {
```

Comment:
For the doc comment: can you call this method before calling SetCardID?

nit: for consistency with other APIs in this package, return a pointer

```
func (yk *YubiKey) CardID() (*CardID, error)
```

Reply:
agreed